packages/twenty-server/src/engine/twenty-orm/utils/build-permission-options.util.ts

```diff
@@ -0,0 +1,18 @@
+import { type RoleContext } from 'src/engine/metadata-modules/role/types/role-context.type';
+
+export const buildPermissionOptions = (
+  roleContext: RoleContext,
+):
+  | { roleId: string }
+  | { roleIds: { intersection: string[] } }
```

Comment:
Took me a while to understand this api, before I realized I suggested it, so maybe it would be nice to add `union` and leaving it with a TODO/throwing an exception, that way the reader would understand what is the intent here? We can also create a dedicated type for this as it's redeclared multiple times